File: sdks/go/cmd/elydora/plugins/claudecode_official_test.go
package plugins

import (
	"os"
	"os/exec"
	"strings"
	"testing"
)

func TestClaudeOfficialBinaryAcceptsInstalledSettings(t *testing.T) {
	binary := os.Getenv("ELYDORA_CLAUDE_BINARY")
	if binary == "" {
		t.Skip("ELYDORA_CLAUDE_BINARY is unset")
	}
	source := `{"hooks":{
  "Stop":[{"hooks":[{
    "type":"command",
    "command":"node",
    "args":["--version"],
    "asyncRewake":true,
    "rewakeMessage":"Background validation failed",
    "rewakeSummary":"Validation feedback"
  }]}]
}}`
	fixture := prepareClaudeFixture(
		t,
		claudeFixtureOptions{existingRaw: &source},
	)
	if err := fixture.plugin.Install(fixture.config); err != nil {
		t.Fatalf("install Claude hooks: %v", err)
	}
	environment := append(
		os.Environ(),
		"HOME="+fixture.homeDir,
		"USERPROFILE="+fixture.homeDir,
		"CLAUDE_CONFIG_DIR="+fixture.configDir,
		"DISABLE_AUTOUPDATER=1",
		"DISABLE_TELEMETRY=1",
	)
	version := exec.Command(binary, "--version")
	version.Dir = fixture.projectDir
	version.Env = environment
	versionOutput, err := version.CombinedOutput()
	if err != nil || !strings.Contains(string(versionOutput), "Claude Code") {
		t.Fatalf("official Claude version: %v\n%s", err, versionOutput)
	}
	doctor := exec.Command(binary, "doctor")
	doctor.Dir = fixture.projectDir
	doctor.Env = environment
	doctorOutput, err := doctor.CombinedOutput()
	if err != nil {
		t.Fatalf("official Claude doctor: %v\n%s", err, doctorOutput)
	}
	output := strings.ToLower(string(doctorOutput))
	for _, problem := range []string{
		"invalid settings",
		"settings validation",
		"failed to parse",
	} {
		if strings.Contains(output, problem) {
			t.Fatalf("official Claude doctor reported %q:\n%s", problem, doctorOutput)
		}
	}
}

File: sdks/node/test/claudecode-official.test.mjs
import assert from 'node:assert/strict';
import test from 'node:test';
import {
  createFixture,
  runProcess,
} from '../test-support/claudecode-test-helpers.mjs';

const claudeBinary = process.env.ELYDORA_CLAUDE_BINARY;

test('official Claude Code accepts the installed user hook contract', {
  skip: claudeBinary ? false : 'set ELYDORA_CLAUDE_BINARY to the official Claude Code executable',
}, async () => {
  const fixture = await createFixture({
    settings: {
      hooks: {
        Stop: [{
          hooks: [{
            type: 'command',
            command: process.execPath,
            args: ['--version'],
            asyncRewake: true,
            rewakeMessage: 'Background validation failed',
            rewakeSummary: 'Validation feedback',
          }],
        }],
      },
    },
  });
  try {
    assert.equal((await fixture.install()).code, 0);
    const environment = {
      HOME: fixture.homeDir,
      USERPROFILE: fixture.homeDir,
      CLAUDE_CONFIG_DIR: fixture.claudeConfigDir,
      DISABLE_AUTOUPDATER: '1',
      DISABLE_TELEMETRY: '1',
    };
    const version = await runProcess(
      claudeBinary,
      ['--version'],
      environment,
      fixture.projectDir,
    );
    assert.equal(version.code, 0, version.stderr);
    assert.match(version.stdout, /Claude Code/i);

    const doctor = await runProcess(
      claudeBinary,
      ['doctor'],
      environment,
      fixture.projectDir,
    );
    assert.equal(doctor.code, 0, doctor.stderr);
    const output = `${doctor.stdout}\n${doctor.stderr}`;
    assert.doesNotMatch(output, /invalid settings|settings validation|failed to parse/i);
  } finally {
    await fixture.close();
  }
});
